daml-lf/data/src/test/scala/com/digitalasset/daml/lf/data/RefTest.scala

```diff
@@ -371,19 +374,16 @@ class RefTest extends AnyFreeSpec with Matchers with TableDrivenPropertyChecks w
       validUserIds.foreach(userId => ApplicationId.fromString(userId) shouldBe a[Right[_, _]])
     }
 
-    "reject invalid user ids" in {
-      val invalidCharacters = "àáABCDEFGHIJKLMNOPQRSTUVWXYZ \\%&*()=[]{};<>,?\""
-      val invalidUserIds =
-        invalidCharacters.map(c => c.toString) ++
-          Vector(
-            "john/doe",
-            "JOHn_doe",
-            "",
-            "office365|10030000838D23AF@MicrosoftOnline.com",
-          )
+    "reject user ids containing invalid characters" in {
+      val invalidCharacters = "àá \\%&*()=[]{};<>,?\""
+      val invalidUserIds = invalidCharacters.map(_.toString) :+ "john/doe"
```

Comment:
Isn't `john/doe` on its own invalid in all cases? so code that only detected `/` would pass the test?

Reply:
you are right, that's why the empty string and the office thingy were essential before